kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -395,24 +397,41 @@ public D createOrReplaceWithNew() throws KubernetesClientException {
 
   @Override
   public T createOrReplace(T... items) {
-    T item = getItem();
+    T itemToCreateOrReplace = getItem();
     if (items.length > 1) {
       throw new IllegalArgumentException("Too many items to create.");
     } else if (items.length == 1) {
-      item = items[0];
+      itemToCreateOrReplace = items[0];
     }
 
-    if (item == null) {
+    if (itemToCreateOrReplace == null) {
       throw new IllegalArgumentException("Nothing to create.");
     }
 
-    if (Utils.isNullOrEmpty(name) && item instanceof HasMetadata) {
-      return withName(((HasMetadata)item).getMetadata().getName()).createOrReplace(item);
+    if (Utils.isNullOrEmpty(name)) {
+
+      return withName(itemToCreateOrReplace.getMetadata().getName()).createOrReplace(itemToCreateOrReplace);
```

Comment:
I'm guessing this recurses into this function

Reply:
Yes, I think previously also it was like this.